File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> mp;
        for(int num : nums){
            mp[num]++;
        }
        int maxfreq = 0;
        int maxele = 0;
        for(auto it : mp){
            if(it.second > maxfreq){
                maxfreq = it.second;
                maxele = it.first;
            }
        }
        return maxele;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq = {}
        maxele = 0
        maxfreq = 0
        for i in range(len(nums)):
            freq[nums[i]] = freq.get(nums[i], 0) + 1
        for a, b in freq.items():
            if b > maxfreq:
                maxfreq = b
                maxele = a
        return maxele

        